packages/admin-app/src/components/ElectionSetupWizard/Steps/JointKeyUploadStep.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { CiphertextElectionContext, SubmitElectionRequest } from '@electionguard/api-client';
+import IconHeader from '../../IconHeader';
+import { Message, MessageId } from '../../../lang';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        flexGrow: 1,
+        height: '100%',
+    },
+    wrapper: {
+        margin: theme.spacing(1),
+        position: 'relative',
+    },
+    content: {
+        paddingTop: theme.spacing(4),
+        paddingBottom: theme.spacing(4),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    spaced: {
+        marginBottom: theme.spacing(2),
+    },
+    buttonProgress: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        marginTop: -12,
+        marginLeft: -12,
+    },
+    error: {
+        color: 'red',
+        marginBottom: theme.spacing(2),
+    },
+}));
+
+export interface JointKeyUploadStepProps {
+    onNext: () => void;
+    onChanged: (newSubmitElectionRequest: SubmitElectionRequest) => void;
+}
+
+/**
+ * Joint Key Select Step for Election Setup
+ */
+const JointKeyUploadStep: React.FC<JointKeyUploadStepProps> = ({ onNext, onChanged }) => {
+    const [uploading, setUploading] = useState(false);
+    const [error, setError] = useState<string>();
+    const classes = useStyles();
+    const intl = useIntl();
+
+    const setIntlError = (id: string) => {
+        const message = intl.formatMessage({
+            id,
+        });
+        setError(message);
+    };
+
+    const onFileUpload = async (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (e?.target?.files?.length) {
+            setUploading(true);
+            const file = e.target.files[0];
+            const text = await file.text();
+            try {
+                const keys = JSON.parse(text);
+                const context = {
+                    commitment_hash: keys.commitment_hash,
+                    crypto_base_hash: keys.crypto_base_hash,
+                    crypto_extended_base_hash: keys.crypto_extended_base_hash,
+                    elgamal_public_key: keys.elgamal_public_key,
+                } as CiphertextElectionContext;
+                const fileValid =
+                    context.commitment_hash &&
+                    context.crypto_base_hash &&
+                    context.crypto_extended_base_hash &&
+                    context.elgamal_public_key;
+                if (!fileValid) {
+                    throw new Error();
+                }
+                onChanged({ context } as SubmitElectionRequest);
+                onNext();
+            } catch (ex) {
+                setIntlError(MessageId.ElectionSetup_JointKeyUpload_InvalidFile);
+            } finally {
+                setUploading(false);
+            }
+        } else {
+            setIntlError(MessageId.ElectionSetup_JointKeyUpload_NoFile);
```

Comment:
I tried, but couldn't actually get this line to execute.  Not sure how you could get into this method with zero files, but I put it in just in case there was some odd edge case I was missing

Reply:
I like having this here in case of any part of the data getting passed in is null.